---
title: 'curatedTCGA metadata: LUSC'
author: "Britney Pheng"
date: "`r Sys.Date()`"
output: html_document
---
# Load Packages
```{r load packages, warning=FALSE, message=FALSE}
suppressPackageStartupMessages({
  library(GenomicSuperSignature)
  library(curatedTCGAData)
  library(MultiAssayExperiment)
  library(TCGAutils)
  library(tidyr)
  library(dplyr)
  library(ggplot2)
  library(magick)
  library(wordcloud)
  library(EnrichmentBrowser)
})
```

```{r RAVmodel, echo=FALSE, warning=FALSE, message=FALSE}
RAVmodel <- getModel("C2", load=TRUE)
```

# TCGA Datasets
```{r load datasets}
load("~/Documents/GitHub/GSS/data/TCGA_validationDatasets.rda")
datasets <- TCGA_validationDatasets[1:7]
#head(datasets)
```

# Overview
## heatmapTable (all)
```{r datasets heatmap table}
## heatmapTable
val_all <- validate(datasets, RAVmodel)
heatmapTable(val_all, RAVmodel, scoreCutoff = 0.7)
```
## heatmapTable: LUSC
```{r}
val_lusc <- validate(datasets[['LUSC']], RAVmodel)
heatmapTable(val_lusc, RAVmodel)
```
# curated TCGA Data for LUSC

## Select table of LUSC RNA metadata
```{r MetaData, message=FALSE}
tcga_lusc <- curatedTCGAData(diseaseCode = 'LUSC', assays = "RNA*", version = "2.0.1", dry.run = FALSE)
```
## Select for 'LUSC_RNASeq2Gene-20160128' metadata with dimensions (423 x 3)
```{r Select SummarizedExperiment assay, warning=FALSE}
(tcga_lusc_rna <- getWithColData(tcga_lusc, "LUSC_RNASeq2Gene-20160128", mode = "append"))
```

```{r}
## use colData()
length(colnames(colData(tcga_lusc_rna))) #2258 gene columns

ind <- sample(1:2258, 50)
test_dat <- colData(tcga_lusc_rna)[ind] %>% as.matrix
calculateScore(test_dat, RAVmodel)

## no colData()
sampleScore <- calculateScore(tcga_lusc_rna, RAVmodel)

val_all <- validate(tcga_lusc_rna, RAVmodel)
validated_ind <- validatedSignatures(val_all, num.out = 10, RAVmodel,
                                     swCutoff = 0, indexOnly = TRUE)
validated_ind
```


## heatmapTable (curatedTCGAData - LUSC)
```{r tcga, message = FALSE}
val_tcga_rna <- validate(tcga_lusc_rna, RAVmodel)
heatmapTable(val_tcga_rna, RAVmodel)
```

Compared to the prior heatmap tables, RAV 814 is the only overlapping RAV.

## RAV 814
```{r echo = FALSE, warning = FALSE}
ind <- 814
findStudiesInCluster(RAVmodel, ind, studyTitle = TRUE)
subsetEnrichedPathways(RAVmodel, ind, include_nes = TRUE) %>% as.data.frame
drawWordcloud(RAVmodel, ind)
```


